src/adaptation/icap/ModXact.cc

```diff
@@ -1157,6 +1157,31 @@ void Adaptation::Icap::ModXact::decideOnParsingBody()
     }
 }
 
+namespace Adaptation {
+namespace Icap {
+class ExtensionsParser : public Http1::CustomExtensionsParser
+{
+public:
+    ExtensionsParser() : useOriginBody_(-1) {}
+
+    virtual bool parse(Tokenizer &tok, const SBuf &extName) override {
+        assert(knownExtension(extName));
+        return parseIntExtension(tok, UseOriginalBodyName, useOriginBody_);
+    }
+    virtual bool knownExtension(const SBuf &extName) override { return extName == UseOriginalBodyName;}
+    int64_t useOriginBody() const { return useOriginBody_; }
```

Comment:
Please split into two methods to hide our internal hack of representing abcent extension as a negative offset:

```suggestion
    bool sawUseOriginalBody() const { return useOriginalBody_ >= 0; }
    uint64_t useOriginalBody() const { assert(sawUseOriginalBody()); return static_cast<uint64_t>(useOriginalBody_); }
```